### DISCUSSION TITLE
Use ESP-IDF UART features

### DISCUSSION THREAD

**Discussion by bertmelis** [eModbus/eModbus#102] [Ideas]
I haven't looked into it yet, but there has been a PR merged to build ESP32 Arduino HardwareSerial on top of ESP-IDF: https://github.com/espressif/arduino-esp32/pull/5549

This might give us a possibility to simplify the RTU code. ESP-IDF has way more options like buffer configuration  and silent time.

Mind: it is merged in `master`. Not yet released.

**Comment by Miq1** [comment]
Interesting, but I would suggest to wait a bit to see if others will find issues... I do not want to be a guinea pig for the new code, since our lib is at least working okay as is.

**Reply by bertmelis** [reply]
Off course! There's absolutely no need to rush.

**Reply by Miq1** [reply]
Was the Platformio ESP32 platform updated to the 2.0.0 release already? Then we should be able to see if the issues are different that we have encountered recently.

**Reply by bertmelis** [reply]
I actually never understood how Platformio updates their stuff. Do they just fetch the latest release or do they have to change the version manually?

In any case: the Arduino framework for ESP32 has been released with these changes..

EDIT: I guess not yet supported: https://github.com/platformio/platform-espressif32/issues/619
